core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -4273,18 +4386,10 @@ private long o3ScheduleMoveUncommitted0(int timestampIndex, long transientRowsAd
                 }
             }
 
-            for (int n = o3PendingCallbackTasks.size() - 1; n > -1; n--) {
-                final O3CallbackTask task = o3PendingCallbackTasks.getQuick(n);
-                if (task.tryLock()) {
-                    O3CallbackJob.runCallbackWithCol(
-                            task,
-                            -1,
-                            null
-                    );
-                }
-            }
-
-            o3DoneLatch.await(queuedCount);
+            dispatchO3CallbackQueue(queue, queuedCount);
+        }
+        if (o3ErrorCount.get() > 0) {
```

Comment:
Do we need this check here considering that `dispatchO3CallbackQueue` has a `checkO3Errors` call?

Reply:
removed